projects/vdk-core/tests/vdk/internal/builtin_plugins/ingestion/test_ingester_router.py

```diff
@@ -76,48 +77,31 @@ def test_router_send_tabular_data_for_ingestion_no_default_method(
 
 
 @patch(f"{IngesterRouter.__module__}.{IngesterBase.__name__}", spec=IngesterBase)
-def test_router_use_chained_ingest_plugins_send_object(
-    mock_ingester_base: MagicMock,
-):
-    router = create_ingester_router({"ingest_method_default": "test-ingest"})
-
-    router.add_ingester_factory_method(
-        "pre-ingest-test", lambda: MagicMock(spec=IIngesterPlugin)
-    )
-    router.add_ingester_factory_method(
-        "test-ingest", lambda: MagicMock(spec=IIngesterPlugin)
-    )
+def test_router_no_chained_ingest_plugins(mock_ingester_base: MagicMock):
+    router = create_ingester_router({"ingest_method_default": "test"})
+    router.add_ingester_factory_method("test", lambda: MagicMock(spec=IIngesterPlugin))
 
     router.send_object_for_ingestion({"a": "b"})
 
     mock_ingester_base.return_value.send_object_for_ingestion.assert_called_with(
-        {"a": "b"}, None, "test-ingest", None, None
+        {"a": "b"}, None, "test", None, None
     )
-    # TODO: Uncomment when implementation is ready
-    # mock_ingester_base.return_value.send_tabular_data_for_ingestion.assert_called_with(
-    #     ["b"], ["a"], None, "pre-ingest-test", None, None
-    # )
 
+    assert router._pre_processor_sequence == []
+    assert router._post_processor_sequence == []
 
-@patch(f"{IngesterRouter.__module__}.{IngesterBase.__name__}", spec=IngesterBase)
-def test_router_use_chained_ingest_plugins_tabular_data(
-    mock_ingester_base: MagicMock,
-):
-    router = create_ingester_router({"ingest_method_default": "test-ingest"})
 
-    router.add_ingester_factory_method(
-        "pre-ingest-test", lambda: MagicMock(spec=IIngesterPlugin)
-    )
-    router.add_ingester_factory_method(
-        "test-ingest", lambda: MagicMock(spec=IIngesterPlugin)
-    )
+def test_router_raise_error_chained_ingest_plugins_not_registered():
+    config = {
+        "ingest_payload_preprocess_sequence": "pre-ingest-test",
+        "ingest_payload_postprocess_sequence": "post-ingest-test",
+        "ingest_method_default": "test",
+    }
 
-    router.send_tabular_data_for_ingestion(rows=["b"], column_names=["a"])
+    router = create_ingester_router(config)
 
-    mock_ingester_base.return_value.send_tabular_data_for_ingestion.assert_called_with(
-        ["b"], ["a"], None, "test-ingest", None, None
-    )
-    # TODO: Uncomment when implementation is ready
-    # mock_ingester_base.return_value.send_tabular_data_for_ingestion.assert_called_with(
-    #     ["b"], ["a"], None, "pre-ingest-test", None, None
-    # )
+    with pytest.raises(VdkConfigurationError):
```

Comment:
maybe add check the processors are now initialised once configured?

Reply:
The processors would not be initialized in this case, as there are no registered plugins.

Reply:
yup, seems missing coverage for scenarios when processors are initialized, pre- and post-